File: benchmark/comparison/results/blind_ts/24.ts
type Option = { tag: "Some"; value: number } | { tag: "None" };

function Some(v: number): Option { return { tag: "Some", value: v }; }
const None: Option = { tag: "None" };

function max2(a: number, b: number): number {
  return a >= b ? a : b;
}

function max_opt(a: Option, b: Option): Option {
  if (a.tag === "Some" && b.tag === "Some") return Some(max2(a.value, b.value));
  if (a.tag === "Some") return a;
  if (b.tag === "Some") return b;
  return None;
}

function formatOpt(o: Option): string {
  if (o.tag === "Some") return `Some(${o.value})`;
  return "None";
}

console.log(max2(3, 7));
console.log(max2(10, 2));
console.log(max2(5, 5));
console.log(formatOpt(max_opt(Some(3), Some(7))));
console.log(formatOpt(max_opt(Some(5), None)));
console.log(formatOpt(max_opt(None, Some(3))));
console.log(formatOpt(max_opt(None, None)));

File: benchmark/comparison/results/blind_py/24.py
class Some:
    def __init__(self, v):
        self.v = v

class NoneOpt:
    pass

def max2(a, b):
    return a if a >= b else b

def max_opt(a, b):
    if isinstance(a, Some) and isinstance(b, Some):
        return Some(max2(a.v, b.v))
    if isinstance(a, Some):
        return a
    if isinstance(b, Some):
        return b
    return NoneOpt()

def fmt_opt(o):
    if isinstance(o, Some):
        return f"Some({o.v})"
    return "None"

print(max2(3, 7))
print(max2(10, 2))
print(max2(5, 5))
print(fmt_opt(max_opt(Some(3), Some(7))))
print(fmt_opt(max_opt(Some(5), NoneOpt())))
print(fmt_opt(max_opt(NoneOpt(), Some(3))))
print(fmt_opt(max_opt(NoneOpt(), NoneOpt())))
